CHANGELOG.md

```diff
@@ -23,6 +23,33 @@
  - update prelude documentation
  - increase precision of metrics for smaller values
  - consistently build configuration from arguments
+ - replace `GooseAttack::set_host()` with more generic `GooseAttack::set_default()`, exposes the following defaults:
+    o default host: `GooseDefault::Host` (&str)
+    o default users to start: `GooseDefault::Users` (usize)
+    o default users to start per second: `GooseDefault::HatchRate` (usize)
+    o default number of seconds for test to run: `GooseDefault::RunTime` (usize)
+    o default log level: `GooseDefault::LogLevel` (usize)
+    o default log file name: `GooseDefault::LogFile` (&str)
+    o default verbosity: `GooseDefault::Verbose` (usize)
+    o default to only printing final summary metrics: `GooseDefault::OnlySummary` (bool)
+    o default to not resetting metrics after all users start: `GooseDefault::NoResetMetrics` (bool)
+    o default to not tracking metrics: `GooseDefault::NoMetrics` (bool)
+    o default to not tracking task metrics: `GooseDefault::NoTaskMetrics` (bool)
+    o default metrics log file name: `GooseDefault::MetricsFile` (&str)
+    o default metrics log file format: `GooseDefault::MetricsFormat` (&str)
+    o default debug log file name: `GooseDefault::DebugFile` (&str)
+    o default debug log file format: `GooseDefault::DebugFormat` (&str)
+    o default to tracking status codes: `GooseDefault::StatusCodes` (bool)
+    o default maximum requests per second: `GooseDefault::ThrottleRequests` (usize)
+    o default to following redirect of base_url: `GooseDefault::StickyFollow` (bool)
+    o default to enabling Manager mode: `GooseDefault::Manager` (bool)
+    o default number of Workers to expect: `GooseDefault::ExpectWorkers` (usize)
+    o default to ignoring load test checksum: `GooseDefault::NoHashCheck` (bool)
+    o default host to bind Manager to: `GooseDefault::ManagerBindHost` (&str)
```

Comment:
I wish all these were their own thing, then maybe `GooseDefault::Manager` could be an `Option<ManagerDefaults>` so when None, Goose does not run in Manager mode.

Reply:
One of the improvements made in `0.10` is a more logical grouping of options in the help text when you run `-h`. Does that address your concern? Or what would be solving by moving Gaggle options into their own struct?

```rust
Options available when launching a Goose load test.


Optional arguments:
  -h, --help                 Displays this help
  -V, --version              Prints version information
  -l, --list                 Lists all tasks and exits

  -H, --host HOST            Defines host to load test (ie http://10.21.32.33)
  -u, --users USERS          Sets concurrent users (default: number of CPUs)
  -r, --hatch-rate RATE      Sets per-second user hatch rate (default: 1)
  -t, --run-time TIME        Stops after (30s, 20m, 3h, 1h30m, etc)
  -g, --log-level            Sets log level (-g, -gg, etc)
  -L, --log-file NAME        Enables log file and sets name
  -v, --verbose              Sets debug level (-v, -vv, etc)

Metrics:
  --only-summary             Only prints final summary metrics
  --no-reset-metrics         Doesn't reset metrics after all users have started
  --no-metrics               Doesn't track metrics
  --no-task-metrics          Doesn't track task metrics
  -m, --metrics-file NAME    Sets metrics log file name
  --metrics-format FORMAT    Sets metrics log format (csv, json, raw)
  -d, --debug-file NAME      Sets debug log file name
  --debug-format FORMAT      Sets debug log format (json, raw)
  --status-codes             Tracks additional status code metrics

Advanced:
  --throttle-requests VALUE  Sets maximum requests per second
  --sticky-follow            Follows base_url redirect with subsequent requests

Gaggle:
  --manager                  Enables distributed load test Manager mode
  --expect-workers VALUE     Sets number of Workers to expect
  --no-hash-check            Tells Manager to ignore load test checksum
  --manager-bind-host HOST   Sets host Manager listens on (default: 0.0.0.0)
  --manager-bind-port PORT   Sets port Manager listens on (default: 5115)
  --worker                   Enables distributed load test Worker mode
  --manager-host HOST        Sets host Worker connects to (default: 127.0.0.1)
  --manager-port PORT        Sets port Worker connects to (default: 5115)
```